### PR TITLE
SD-5002 - highlight multiple spaces in spell check

### PR DESCRIPTION
**Pull request by lnogues**
Hightlight multiple spaces in spellcheck and suggest a single space instead.

### CODE REVIEW

File: scripts/superdesk/editor/spellcheck/spellcheck.js [hunk 1]
```diff
@@ -285,19 +297,26 @@ function SpellcheckService($q, api, dictionaries, $rootScope, $location) {
...
+            }).then(function(result) {
+                var allDict = getDict();
+                var wordFoundInDict = _.pick(allDict.content, function(value, key) {
+                    if (key.toLowerCase() === word.toLowerCase()) {
+                        return key;
+                    }
+                });
+                angular.extend(result.corrections, Object.keys(wordFoundInDict));
+
+                return result.corrections.map(function(key) {
```
Comment (gbbr) [inline scripts/superdesk/editor/spellcheck/spellcheck.js]: You may use ES6 syntax here if you'd like `return result.corrections.map(key => ({key: key, value: key}));`

Comment (lnogues) [inline scripts/superdesk/editor/spellcheck/spellcheck.js]: Well since I didn't see any arrow functions in the file, I kept the old syntax

Comment (gbbr) [inline scripts/superdesk/editor/spellcheck/spellcheck.js]: I know, it's fine. We've just recently started supporting ES6 and I'm trying to encourage it more.

File: scripts/superdesk/editor2/views/block-text.html [hunk 2]
```diff
@@ -20,7 +20,7 @@
     </code>
 
     <ul class="dropdown-menu">
-        <li ng-repeat="word in suggestions"><button ng-click="replace(word)">{{ :: word }}</button></li>
+        <li ng-repeat="word in suggestions"><button ng-click="replace(word.key)">{{ :: word.value }}</button></li>
```
Comment (sjunaid) [inline scripts/superdesk/editor2/views/block-text.html]: Hi @lnogues , there is a change impact in superdesk\editor2\editor.js, 
- the below code need to be adjusted for capital word suggestion,

spellcheck.suggest(node.textContent).then(function(suggestions) {
                            if (scope.sentenceWord) {
                                ```suggestions.push(scope.replaceWord[0].toUpperCase() + scope.replaceWord.slice(1));```
                                scope.suggestions = _.unique(suggestions);
                            } else {

- use _.uniq instead, as _.unique alias is removed in favor of _.uniq in ver. 4. https://github.com/lodash/lodash/wiki/Changelog

File: scripts/superdesk/editor2/editor.js [hunk 3]
```diff
@@ -1151,8 +1151,14 @@ angular.module('superdesk.editor2', [
...
                             if (scope.sentenceWord) {
-                                suggestions.push(scope.replaceWord[0].toUpperCase() + scope.replaceWord.slice(1));
-                                scope.suggestions = _.unique(suggestions);
+                                suggestions.push({
+                                    key: scope.replaceWord[0].toUpperCase() + scope.replaceWord.slice(1),
+                                    value: scope.replaceWord[0].toUpperCase() + scope.replaceWord.slice(1)
+                                });
+
+                                scope.suggestions = suggestions.filter(suggestion => {
+                                    return suggestion.key !== scope.replaceWord;
```
Comment (gbbr) [inline scripts/superdesk/editor2/editor.js]: You don't necesarilly have to change this, as it is already correct, but just FYI, `return` can be omitted here with this form:
```js
scope.suggestions = suggestions.filter(suggestion => suggestion.key !== scope.replaceWord);
```

**Comment by vied12** [comment]
:+1:

**Comment by gbbr** [comment]
👍

**Pull request merged**